From 9581aaf45ec463a5c12730dbe87e2cbfb996de36 Mon Sep 17 00:00:00 2001
From: Feng Pei <quic_fpei@quicinc.com>
Date: Tue, 4 Jun 2024 02:44:06 -0700
Subject: [PATCH] Revert "linux: Bridge fdb add or refresh"

This reverts commit 4996b376e428668ca35b561a74d8ece68b590199.

Reason for revert: This API is not used in 6.6.

Change-Id: I60051dab1d3a127318a5000f6cd88501715a4ef2
---
 include/linux/if_bridge.h |  2 --
 net/bridge/br_fdb.c       | 46 ---------------------------------------
 2 files changed, 48 deletions(-)

diff --git a/include/linux/if_bridge.h b/include/linux/if_bridge.h
index dc8ea2bb9ca9..29b503bbb889 100644
--- a/include/linux/if_bridge.h
+++ b/include/linux/if_bridge.h
@@ -282,8 +282,6 @@ extern void br_fdb_unregister_notify(struct notifier_block *nb);
 extern struct net_device *br_fdb_bridge_dev_get_and_hold(struct net_bridge *br);
 extern int br_fdb_delete_by_netdev(struct net_device *dev,
 			const unsigned char *addr, u16 vid);
-extern int br_fdb_add_or_refresh_by_netdev(struct net_device *dev,
-			    const unsigned char *addr, u16 vid, u16 state);
 
 typedef struct net_bridge_port *br_get_dst_hook_t(
 		const struct net_bridge_port *src,
diff --git a/net/bridge/br_fdb.c b/net/bridge/br_fdb.c
index 6727a6c58265..aa5ae9025e38 100644
--- a/net/bridge/br_fdb.c
+++ b/net/bridge/br_fdb.c
@@ -1402,52 +1402,6 @@ static int __br_fdb_delete(struct net_bridge *br,
 	return err;
 }
 
-/* This function creates a new FDB entry.
- * The caller can specify the FDB entry type like static,
- * local or external entry.
- * This has to be called only for bridge-port netdevs.
- */
-int br_fdb_add_or_refresh_by_netdev(struct net_device *dev,
-				    const unsigned char *addr, u16 vid,
-				    u16 state)
-{
-	struct net_bridge_fdb_entry *fdb = NULL;
-	struct net_bridge *br = NULL;
-	int err = 0;
-	u16 nlh_flags = NLM_F_CREATE;
-	struct net_bridge_port *p = NULL;
-
-	if (!dev) {
-		pr_info("bridge: netdevice is NULL\n");
-		return -EINVAL;
-	}
-
-	rcu_read_lock();
-	p = br_port_get_check_rcu(dev);
-	if (!p) {
-		rcu_read_unlock();
-		pr_info("bridge: %s not a bridge port\n",
-			dev->name);
-		return -EINVAL;
-	}
-
-	br = p->br;
-
-	spin_lock_bh(&br->hash_lock);
-	fdb = br_fdb_find(br, addr, vid);
-	if (!fdb) {
-		err = fdb_add_entry(br, p, addr, state,
-				    nlh_flags, vid, 0);
-	} else {
-		fdb->updated = jiffies;
-	}
-	spin_unlock_bh(&br->hash_lock);
-	rcu_read_unlock();
-
-	return err;
-}
-EXPORT_SYMBOL_GPL(br_fdb_add_or_refresh_by_netdev);
-
 /* This function has to be called only for bridge-port netdevs.*/
 /* For bridge netdev br_fdb_delete has to be called.*/
 int br_fdb_delete_by_netdev(struct net_device *dev,
-- 
2.34.1

